backend/svirt.pm

```diff
@@ -270,35 +293,74 @@ sub start_serial_grab {
     }
 }
 
-# opens another SSH connection to grab the serial console with the specified port
-sub open_serial_console_via_ssh {
-    my ($self, $name, $port) = @_;
+=head2 ($ssh, $chan) = $self->backend->start_serial_grab($name, %args)
 
-    bmwqemu::diag("Starting SSH connection to connect to libvirt domain $name via serial port $port");
-    my $credentials = $self->read_credentials_from_virsh_variables;
-    my $ssh         = $self->new_ssh_connection(%$credentials);
-    my $chan        = $ssh->channel();
-    die 'No channel found' unless $chan;
-    $chan->blocking(0);
-    $chan->pty('vt100', {echo => 1});
-    $chan->pty_size(1024, 24);
-    $chan->shell();
-    print($chan "PS1='# '\n");
+Opens SSH connection to grab the serial console for kernel log (serial0.txt)
+or serial terminal log (serial_terminal.txt).
+
+C<$args{port}> used non-default port (used for serial_terminal.txt)
+C<$args{devname}> used device name (usually console, for serial_terminal.txt)
+C<$args{is_terminal}> for serial terminal (usually console, for serial_terminal.txt)
```

Comment:
Maybe it would be better to mention the console/screen as well as the output file.

Reply:
Fixed.